core/src/test/java/com/google/errorprone/bugpatterns/ArrayToStringTest.java

```diff
@@ -185,4 +185,18 @@ public void positiveConcat() {
   public void negativeConcat() {
     compilationHelper.addSourceFile("ArrayToStringConcatenationNegativeCases.java").doTest();
   }
+
+  @Test
+  public void arrayPassedToJoiner() {
+    compilationHelper
+        .addSourceLines(
+            "Test.java",
+            "import com.google.common.base.Joiner;",
+            "class Test {",
+            "  String test(Joiner j, Object[] a) {",
+            "    return j.join(a);",
+            "  }",
```

Comment:
@graememorgan Do you intend to support the `Joiner.join` overload where the varargs parameter has other fixed parameters before it?

One of the codebases I maintain has this example:
```
    private static String makePath(Namespace namespace, Project project, String... path) {
        return "/" + Joiner.on("/").join(namespace.name(), project.name(), (Object[]) path);
    }
```

And it's now erroring with:

```
error: [ArrayToString] Calling toString on an array does not provide useful information
        return "/" + Joiner.on("/").join(namespace.name(), project.name(), (Object[]) path);
                                                                           ^
    (see https://errorprone.info/bugpattern/ArrayToString)
  Did you mean 'return "/" + Joiner.on("/").join(namespace.name(), project.name(), Arrays.toString((Object[]) path));'?
```

The suggestion is an unwanted behavior change, because it introduces the array format and causes tests to start failing with:

```
Expected :"/NamespacePath/ProjectPath/folder1/folder2/folder3"
Actual   :"/NamespacePath/ProjectPath/[folder1, folder2, folder3]"
```

Reply:
@ash211 thanks for the comment, there's a fix in progress for that